### PR TITLE
fix: field sorting will send two requests in weak network

### PR DESCRIPTION
**Pull request by chavyleung**
## Change Summary

- closes: https://github.com/nocodb/nocodb/issues/8613

## Change type

- [x] fix: (bug fix for the user, not a fix to a build script)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/grid/useColumnDrag.ts [hunk 1]
```diff
@@ -106,9 +107,11 @@ export const useColumnDrag = ({
   }
 
   const handleReorderColumn = async () => {
+    isProcessing.value = true
     dragColPlaceholderDomRef.value!.style.left = '0px'
     dragColPlaceholderDomRef.value!.style.height = '0px'
     await reorderColumn(draggedCol.value!.id!, toBeDroppedColId.value!)
+    isProcessing.value = false
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/components/smartsheet/grid/useColumnDrag.ts:114]: let's wrap it with try...catch and reset `isProcessing` state within final block

**Comment by chavyleung** [comment]
@pranavxc Is there anything I should do next?

**Comment by pranavxc** [comment]
> @pranavxc Is there anything I should do next?

Soon I will do a final review.

File: packages/nc-gui/components/smartsheet/grid/useColumnDrag.ts [hunk 2]
```diff
@@ -180,11 +183,13 @@
   }
 
   // fallback for safari browser
-  const onDragEnd = (e: DragEvent) => {
+  const onDragEnd = async (e: DragEvent) => {
     e.preventDefault()
 
     if (!e.dataTransfer || !draggedCol.value || !toBeDroppedColId.value) return
 
+    await until(() => isProcessing.value == false).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (pranavxc) [reply packages/nc-gui/components/smartsheet/grid/useColumnDrag.ts:194]: we can replace this with 

```javascript
await until(() => !isProcessing.value).toBeTruthy()
```

**Review by pranavxc — APPROVED** [review]

**Pull request merged**